Python/Product/Analysis/LanguageServer/DisplayTextBuilder.cs

```diff
@@ -0,0 +1,147 @@
+        public string MakeHoverText(IEnumerable<AnalysisValue> values, string originalExpression, InformationDisplayOptions displayOptions) {
+            string firstLongDescription = null;
+            var multiline = false;
+            var result = new StringBuilder();
+            var documentations = new HashSet<string>();
+
+            foreach (var v in values) {
+                var doc = GetDocString(v);
+                firstLongDescription = firstLongDescription ?? doc;
+
+                doc = displayOptions.trimDocumentationLines ? LimitLines(doc) : doc;
+                if (string.IsNullOrEmpty(doc)) {
+                    continue;
+                }
+
+                if (documentations.Add(doc)) {
+                    if (documentations.Count > 1) {
+                        if (result.Length == 0) {
+                            // Nop
+                        } else if (result[result.Length - 1] != '\n') {
+                            result.Append(", ");
+                        } else {
+                            multiline = true;
+                        }
+                    }
+                    result.Append(doc);
+                }
+            }
+
+            if (documentations.Count == 1 && !string.IsNullOrEmpty(firstLongDescription)) {
+                result.Clear();
+                result.Append(firstLongDescription);
+            }
+
+            var displayText = result.ToString();
+            if (displayOptions.trimDocumentationText && displayText.Length > displayOptions.maxDocumentationTextLength) {
+                displayText = displayText.Substring(0,
+                    Math.Max(3, displayOptions.maxDocumentationTextLength) - 3) + "...";
+
+                result.Clear();
+                result.Append(displayText);
+            }
+
+            if (!string.IsNullOrEmpty(originalExpression)) {
+                if (displayOptions.trimDocumentationText && originalExpression.Length > displayOptions.maxDocumentationTextLength) {
+                    originalExpression = originalExpression.Substring(0, 
+                        Math.Max(3, displayOptions.maxDocumentationTextLength) - 3) + "...";
+                }
+                if (multiline) {
+                    result.Insert(0, $"{originalExpression}:{Environment.NewLine}");
+                } else if (result.Length > 0) {
+                    result.Insert(0, $"{originalExpression}: ");
+                } else {
+                    result.Append($"{originalExpression}: <unknown type>");
+                }
+            }
+
+            return _textConverter.ToMarkdown(result.ToString());
+        }
+
+        public string MakeModuleHoverText(ModuleReference modRef) {
+            // Return module information
+            var contents = "{0} : module".FormatUI(modRef.Name);
+            if (!string.IsNullOrEmpty(modRef.Module?.Documentation)) {
+                contents += $"{Environment.NewLine}{Environment.NewLine}{modRef.Module.Documentation}";
```

Comment:
Do we need `FormatUI` here as well?

Reply:
It just does `CurrentCulture`... I am actually not sure that current culture is right.  Here we should have invariant I believe. I.e. if we have programming language docs then numbers must remain `5.0` and should not be displayed as `5,0` in a culture-specific manner.